client/app/controllers/collection/add.js

```diff
@@ -1,125 +1,145 @@
-
-        function condition_dispatcher(condition) {
-
-            // Check if its a regular condition
-            if (condition.parameter !== undefined) {
-                // actualy check the condition is met;
-                // the parameter has to have the given state.
-                if (state[condition.parameter] === undefined) {
-                    state[condition.parameter] = {};
-                }
-                if (state[condition.parameter].state === undefined) {
-                    state[condition.parameter].state = [];
-                }
-                const parameter_state = state[condition.parameter].state
-                // check that the state exists for this item
-                return parameter_state.some((state_item) => state_item === condition.state)
-            }
-
-            // Check if its an 'all' composite condition
-            if (condition.all !== undefined &&
-                condition.all.constructor === Array
-            ) {
-                // if any conditions fail, the whole check fails.
-                return check_all(condition.all);
-            }
-
-            // Check if its an 'any' composite condition
-            if (condition.any !== undefined &&
-                condition.any.constructor === Array
-            ) {
-                // if any conditions are met, the whole check passes.
-                return check_any(condition.any);
-            }
-
-            // Check if its a 'none' composite condition
-            if (condition.none !== undefined &&
-                condition.none.constructor ===Array
-            ) {
-                // if any conditions are met, the whole check fails.
-                return !check_any(condition.none);
-            }
-
-            return false;
-
-        }
-
-        function check_all(conditions) {
-            // if any conditions fail, the whole check fails.
-            return !conditions.some(function(condition) {
-                return !condition_dispatcher(condition);
-            });
-        }
-
-        function check_any(conditions) {
-            // if any conditions are met, the whole check passes.
-            return action.conditions.some(condition_dispatcher)
-        }
-
-        // check if the action can fire.
-        const may_fire = check_all(action.conditions);
-        if (!may_fire) { return; }
+export default Ember.Controller.extend({
 
-        // action may fire
-        this.set('state.'+action.output, {
-            value: this.get(action.type).call(this, (action.parameters)),
-            state: ['defined']
-        });
-    });
-}
 
-export default Ember.Controller.extend({
     addMethod: 'select', // 'select' or 'create'
     methodSelected: false,
     type: Ember.computed('model.settings', function() {
         var collectionType = this.get('model.settings.collectionType') || 'project';
         return collectionType.toLowerCase();
     }),
 
+
     widgets: [],
+    actions: [],
```

Comment:
`actions` is defined in the controller here as well but there should only be one: https://github.com/cos-labs/collections/pull/81/files#diff-0fd60a923647a8bdfca3e3418a32795dR143